es download url below 6.3.0 version is not correct

- rally version: 1.0.1

when rally download es, it's url is "http://xxxx/elasticsearch-oss-6.2.2.tar.gz", but correct url should be "http://xxxx/elasticsearch-6.2.2.tar.gz".  The former url is not applied to versions below 6.3.0.

@danielmitterdorfer  Sorry for the incomplete description.

The complete situation is(https://esrally.readthedocs.io/en/stable/recipes.html): I have two machines, one is coordinator(10.8.18.13), the other is candidate(10.8.18.45).  Then I executed below commands:
 
```
esrallyd start --node-ip=10.8.18.13 --coordinator-ip=10.8.18.13   # on coordinator

esrallyd start --node-ip=10.8.18.45 --coordinator-ip=10.8.18.13   # on candidate
```

Then start esrally on coordinator:

```
esrally --track=pmc --distribution-version=6.2.2 --target-hosts=10.8.18.45:9200 --offline --report-file=~/report4.md --report-format=csv --user-tag="0917:report4"
```

Error occured:

```
[allan@108413 ~]$ esrally --track=pmc --distribution-version=6.2.2 --target-hosts=10.8.18.45:9200 --offline --report-file=~/report8.md --report-format=csv --user-tag="0918:report8"

    ____        ____
   / __ \____ _/ / /_  __
  / /_/ / __ `/ / / / / /
 / _, _/ /_/ / / / /_/ /
/_/ |_|\__,_/_/_/\__, /
                /____/

[INFO] Preparing for race ...
[ERROR] Cannot race. ('Cannot download Elasticsearch distribution from [https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-6.2.2.tar.gz]. Please check that the specified version [6.2.2] is correct.', None)
	Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.6/site-packages/esrally/mechanic/supplier.py", line 303, in fetch
    net.download(download_url, distribution_path, progress_indicator=progress)
  File "/opt/anaconda3/lib/python3.6/site-packages/esrally/utils/net.py", line 61, in download
    raise urllib.error.HTTPError(url, r.status, "", None, None)
urllib.error.HTTPError: HTTP Error 404:

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/anaconda3/lib/python3.6/site-packages/esrally/mechanic/mechanic.py", line 546, in receiveMsg_StartNodes
    nodes = self.mechanic.start_engine()
  File "/opt/anaconda3/lib/python3.6/site-packages/esrally/mechanic/mechanic.py", line 678, in start_engine
    binaries = self.supply()
  File "/opt/anaconda3/lib/python3.6/site-packages/esrally/mechanic/supplier.py", line 167, in __call__
    supplier.fetch()
  File "/opt/anaconda3/lib/python3.6/site-packages/esrally/mechanic/supplier.py", line 309, in fetch
    "version [%s] is correct." % (download_url, self.version))
esrally.exceptions.SystemSetupError: ('Cannot download Elasticsearch distribution from [https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-6.2.2.tar.gz]. Please check that the specified version [6.2.2] is correct.', None)


Getting further help:
*********************
* Check the log files in /allan/.rally/logs for errors.
* Read the documentation at https://esrally.readthedocs.io/en/1.0.0/
* Ask a question on the forum at https://discuss.elastic.co/c/elasticsearch/rally
* Raise an issue at https://github.com/elastic/rally/issues and include the log files in /allan/.rally/logs.

-------------------------------
[INFO] FAILURE (took 7 seconds)
-------------------------------
```

After that, I tried to download elasticsearch 6.2.2  from official, and found the url is "http://xxxx/elasticsearch-6.2.2.tar.gz" instead of "http://xxxx/elasticsearch-oss-6.2.2.tar.gz".

Then I edit the `/opt/anaconda3/lib/python3.6/site-packages/esrally/mechanic/supplier.py` (about line 302)，and hard code the url, it works.

```
                progress = net.Progress("[INFO] Downloading Elasticsearch %s" % self.version)
                download_url='https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-6.2.2.tar.gz'    # hardcode
                net.download(download_url, distribution_path, progress_indicator=progress)
```

--- a/esrally/utils/repo.py
+++ b/esrally/utils/repo.py
@@ -35,7 +35,7 @@ class RallyRepository:
 
     def update(self, distribution_version):
         try:
-            if self.remote and not self.offline:
+            if self.remote:
                 branch = versions.best_match(git.branches(self.repo_dir, remote=self.remote), distribution_version)
                 if branch:
                     # Allow uncommitted changes iff we do not have to change the branch

--- a/tests/utils/repo_test.py
+++ b/tests/utils/repo_test.py
@@ -1,3 +1,4 @@
+import random
 from unittest import TestCase
 import unittest.mock as mock
 
@@ -112,7 +113,7 @@ class RallyRepositoryTests(TestCase):
             root_dir="/rally-resources",
             repo_name="unit-test",
             resource_name="unittest-resources",
-            offline=False)
+            offline=random.choice([True, False]))
 
         r.update(distribution_version="1.7.3")
 

